internal/pkg/feature/feature.go

```diff
@@ -0,0 +1,38 @@
+package feature
+
+import (
+	"context"
+)
+
+type key int
+
+const flaggerKey key = iota
+
+// Flagger returns flag values.
+type Flagger interface {
+	// FlagValue returns the value for a given flag.
+	FlagValue(ctx context.Context, flag Flag) interface{}
+}
+
+// Inject will inject the Flagger into the context.
+func Inject(ctx context.Context, flagger Flagger) context.Context {
+	return context.WithValue(ctx, flaggerKey, flagger)
+}
+
+// GetFlagger returns the Flagger for this context or the default flagger.
+func GetFlagger(ctx context.Context) Flagger {
```

Comment:
If I understand it correctly, this is where downstream consumers can inject their context so flux can use the same flagger. Does that mean we can configure flux feature flags the same way we do with idpe? How would we configure things in OSS? Or are we not concerned about that?

Reply:
This is where we retrieve the feature flagger so we can access them within Flux. I think this method isn't exposed to external packages.

A downstream consumer would inject their flagger into the context using `Inject()`.

I've been a bit confused over the best way to configure flux feature flags in idpe. My current thought is just to duplicate the feature flag directly into the `flags.yml` file for each of the products. In the future, we can potentially merge the flags because flux exposes its flags in `dependencies/feature` with the `All()` method.

For OSS, feature flags get configured with a command line parameter called `--feature-flags`. They're exposed, but they are server-wide and there's no way to do the kind of detailed flagging that we can do in cloud.

Reply:
That all makes sense to me. Thanks for clarifying!